app/Plugin/EnvSource/Model/EnvSourceBackend.php

```diff
@@ -126,9 +126,34 @@ protected function resultToOrgIdentity($result) {
     $orgdata['EmailAddress'] = array();
     
     if($result['env_mail']) {
-      $orgdata['EmailAddress'][0]['mail'] = $result['env_mail'];
-      $orgdata['EmailAddress'][0]['type'] = EmailAddressEnum::Official;
-      $orgdata['EmailAddress'][0]['verified'] = true;
+      if(!empty($this->pluginCfg['sp_type'])) {
+        $delimiter = "";
+        if($this->pluginCfg['sp_type'] == AuthProviderEnum::Shibboleth) {
+          $delimiter = ";";
+        } elseif($this->pluginCfg['sp_type'] == AuthProviderEnum::Simplesamlphp) {
+          $delimiter =",";
+        } elseif($this->pluginCfg['sp_type'] == AuthProviderEnum::Other) {
+          $delimiter =  trim($this->pluginCfg['other_type']);
```

Comment:
If type is `Other` we simply don't support multiple values, and we don't try to `explode()` the value. This also simplifies the user interface, below.

Reply:
Done